File: it.unibo.bls19Local/src/main/java/it/unibo/bls/devices/mock/LedMock.java
package it.unibo.bls.devices.mock;

import it.unibo.bls.interfaces.ILed;

public class LedMock implements ILed{
protected boolean state = false;

//Factory method
public static ILed createLed(){
	return new LedMock();
}
@Override
public void turnOn(){
	state = true;
	showState();
}
@Override
public void turnOff() {
	state = false;
	showState();
}
@Override
public boolean getState(){
	return state;
}

private void showState(){
	System.out.println("Led state=" + state);
}
}

File: it.unibo.bls19Local/src/main/kotlin/it/unibo/bls/kotlin/devices/mock/LedMock.kt
package it.unibo.bls.kotlin.devices.mock

import it.unibo.bls.kotlin.interfaces.ILed

class LedMock : ILed {
    private var state = false
    override fun turnOn() {
        state = true
        showState()
    }

    override fun turnOff() {
        state = false
        showState()
    }

    override fun getState(): Boolean {
        return state
    }

    private fun showState() {
        println("Led state=$state")
    }

    companion object {
        //Factory method
        fun createLed(): ILed {
            return LedMock()
        }
    }
}
